### PR TITLE
add schemars support

### PR DESCRIPTION
**Pull request by Gottox**
This pull request adds optional support for schemars.

### CODE REVIEW

**Comment by kilork** [comment]
Hi @Gottox, thanks for this patch!

You need to modify `examples/gen.rs`. `src/types.rs` is auto-generated.

It is generated using following commands:

```
cargo run --example gen -- types > src/types2.rs
mv src/types2.rs src/types.rs
```

See: [update.sh](https://github.com/kilork/keycloak/blob/master/update.sh#L62)

**Comment by Gottox** [comment]
Thanks! Will adapt! ☺️

**Comment by Gottox** [comment]
* [ ] add code generation
* [ ] document feature flags

@kilork 
How do you want me to document the feature flag?

**Comment by kilork** [comment]
> Thanks! Will adapt! ☺️

Great, sorry, just found time to check. Everything looks fine.

Feature description in `rust` docs. I do not know, maybe there are some good practice out there. I can give example how we do this in other projects, maybe it is not the best way, but at least something: https://github.com/kilork/openid#features

Just add short paragraph about your use case or one sentence and feature flag mentioned. This would be quite useful.

We have same `features` section, it is missing any features right in the moment. Small note: source file for documentation is in the `templates\README.md`.

File: examples/gen.rs [hunk 1]
```diff
@@ -325,7 +329,7 @@ fn write_types(enums: &[EnumType], structs: &[Rc<StructType>]) {
             println!("    pub {}: {},", field.field_name, field_type);
         }
 
-        println!("}}\n");
+        println!("}}");
```
Comment (kilork) [inline examples/gen.rs:332]: Should not be changed, althought we run `cargo fmt` afterwards.

Comment (Gottox) [reply examples/gen.rs]: :heavy_check_mark:

**Pull request merged**